src/server/cluster/incoming_slot_migration.cc

```diff
@@ -192,12 +192,21 @@ bool IncomingSlotMigration::Join(long attempt) {
       return false;
     }
 
-    if ((bc_->WaitFor(absl::ToInt64Milliseconds(timeout - passed) * 1ms)) &&
-        (std::all_of(shard_flows_.begin(), shard_flows_.end(),
-                     [&](const auto& flow) { return flow->GetLastAttempt() == attempt; }))) {
+    // if data was sent after LSN, WaitFor() always returns false so to reduce wait time
+    // we check current state and if WaitFor false but GetLastAttempt() == attempt
+    // the Join is failed and we can return false
```

Comment:
Is this really true? I enter the body of the `while(true)` for the first time and I call 

`wait_res = bc_->WaitFor(wait_time)` and it returns false because it got notified. Now I check with `std::_all_of` and it returns `true`. Why did the migration fail here ? Isn't that the happy path ?